src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -59,6 +59,14 @@ typedef struct AttributeEquivalenceClass
 	Index unionQueryPartitionKeyIndex;
 } AttributeEquivalenceClass;
 
+typedef struct FindRteIdentityContext
+{
+	uint32 targetRTEIdentity;
```

Comment:
do we really need `found` and `tempQuery`? Can't we always use `query` and found would mean `query != NULL`?

Reply:
minor: rte identity is `int` although it cannot be negative.

Reply:
I removed `found` field, but I think we need at least 2 fields since we need to know:
- we found the query
- what the query that this RTE resides in is

If you think we don't need one of them, let me know

Reply:
If ` the query that this RTE resides in is` is NULL can't we call it as not found, and otherwise found?

Reply:
Yeah but we first need to set the `query` field so that when looking at a single `RTE` we know its parent query. So the query will need to be set before we come to the final call which just has an RTE node. 

And while setting the query, we don't know if we already found the `query` we are looking at without an extra field.